x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRuleValidator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.rules;
+
+import org.elasticsearch.xpack.searchbusinessrules.PinnedQueryBuilder;
+
+import java.util.Objects;
+
+public class QueryRuleValidator {
+
+    public static boolean validRule(QueryRule rule) {
+        if (Objects.requireNonNull(rule.type()) == QueryRule.QueryRuleType.PINNED) {
+            return validPinnedRule(rule);
+        }
+        return false;
+    }
+
+    private static boolean validPinnedRule(QueryRule rule) {
```

Comment:
is there a reason why we don't reuse the `validate` method from `QueryRule` instead of having this method?

Reply:
You're right, this is redundant. Thanks